Node/core/src/bots/ChatConnector.ts

```diff
@@ -955,8 +1001,22 @@ export class ChatConnector implements IConnector, IBotStorage {
         options.headers['User-Agent'] = USER_AGENT;
     }
 
-    private addAccessToken(options: request.Options, cb: (err: Error) => void): void {
+    private addAccessToken(options: request.Options | any, cb: (err: Error) => void): void {
         if (this.settings.appId && this.settings.appPassword) {
+            if (this.settings.enableSkills) {
+                const credKeys = Object.keys(this.credentialsCache)
+                const matchingKey = credKeys.filter((key: string) => {
+                    return options.url.indexOf(key) >= 0
+                })
+                // get MicrosoftAppCredentials based on options.url and use it to retrieve the token (getToken)
+                if (matchingKey[0]) {
+                    const creds = this.credentialsCache[matchingKey[0]]
+                    return creds.signRequest(options).then(() => {
```

Comment:
the method signature is `void`, but this is trying to return a Promise (from `.then`). We should likely not return anything, but we do need to have a `catch` statement off of `then`.

Reply:
yes noticed this as well. no need to return promise, instead calling callback. added catch clause as well.